### PR TITLE
Adding join links to players patrons guilds pages (#1411)

### LINKED ISSUE #1411: [MetaOS] Adding join links to players, patrons & guilds pages
Right now, the only way to get to the "join" pages is if you go down the landing page & go through the onboarding game.
I think it would make sense to have additional places pointing there & I think it makes most sense to have the players, patrons & guilds leaderboard loop back there.
Here's what I mean.
Currently:
image
After this issue:
(check the new design by Dave below)

The same thing would need to be done on [players](https://my.metagame.wtf/players), [patrons](https://metagame.wtf/patrons) & [guilds](https://metagame.wtf/guilds) pages

[Link to the figma file](https://www.figma.com/file/hAmFBkfNhI7D4HMZ0jOVRV/Upgrade-callout?node-id=0%3A1)

Patrons page should be linking here https://metagame.wtf/community/join/patrons
Guilds page should be linking here https://metagame.wtf/join/guild
Players page should be linking here https://metagame.wtf/play/paths/engaged-octos-path (until the player signup page is made)

### PR DESCRIPTION
**Pull request by HHH-GH**
## Overview

**What features/fixes does this PR include?**
1. Adds Join as a Player/Patron/Guild calls-to-action at the top of those pages
2. Rearranges the layout of the Search and Filters on the Players page so the players list isn't pushed too far down the page by the CTA and search box and filters

**Please provide the GitHub issue number**

Closes #1411

## Implementation
(This is a v2 because I made the original branch into a massive tangle)

There's nothing special about the implementation for the CTAs that were added.

To change the layout of the Search and Filters on the Players page:
- 'Sticky' functionality moved out of `DesktopFilters` to `PlayerFilters` so it could wrap the search input
- Added `SearchIcon` and `FiltersIcon` to the design system
- Changes in appearance/layout depending on the screen size

**Describe technical (nontrivial / non-obvious) parts of your code**
- Making the search and filters go full-width when sticky got a bit hacky trying to avoid a bug. (I made comments in the code)
- I added a search button to go with the search input for larger screen sizes, assuming that not everyone knows that hitting Enter will submit a form
- The filters icon gets a label when the screen is wide enough

## Assets

Screenshots from the players page at various sizes

**Full size - wide enough for the search and filters to go side by side**  

image
---
**Wide screen but not wide enough for the search and filters to go side by side**  

image
---
**The most narrow screenwidth before switching to the mobile view**  

image
---
**Mobile view but wide**  

image
---

**Standard phone**  
image

### CODE REVIEW

**Comment by HHH-GH** [comment]
Changes are designer-approved, now ready for code review ✔

**Comment by HHH-GH** [comment]
Changes are designer-approved, now ready for code review ✔

**Comment by HHH-GH** [comment]
p.s. we're going to do more work on the search UX as a separate issue

File: packages/web/pages/guilds.tsx [hunk 1]
```diff
@@ -22,10 +23,44 @@ const GuildsPage: React.FC<Props> = ({ guilds }) => (
...
     <HeadComponent
       title="MetaGame Guilds"
-      description="MetaGame is a Massive Online Coordination Game! Guilds participating in MetaGame…."
+      description="MetaGame is a Massive Online Coordination Game! Guilds participating in MetaGame…"
       url="https://my.metagame.wtf/guilds"
     />
-    <GuildList {...{ guilds }} />
+
+    {/* VStack is used to make a consistent gap between the Join CTA and the Guilds list */}
+    <VStack maxW="7xl" w="100%" spacing={{ base: 6, md: 8, lg: 12 }} pb={8}>
```
Comment (vidvidvid) [inline packages/web/pages/guilds.tsx:31]: maybe the spacing is increasing unnecessarily and we could keep it at 6. i've taken a look at other pages and it seems inconsistent and pretty big in some cases.
for example: https://frontend-pr-1484-mjhnbmqqna-uk.a.run.app/roles
or: https://frontend-pr-1484-mjhnbmqqna-uk.a.run.app/raids

Comment (vidvidvid) [reply packages/web/pages/guilds.tsx:31]: but we could do this in another PR after talking to the designers

Comment (vidvidvid) [reply packages/web/pages/guilds.tsx:31]: @davort @peth-yursick what do you think about this?

Comment (HHH-GH) [reply packages/web/pages/guilds.tsx:31]: You're talking about this part, yeah?

image

After looking at the pages you linked, yes I agree the spacing could be tighter.

Dave had mentioned reworking the design around the search box as a separate task
-> https://app.dework.xyz/metagame-1/metafest-2022?taskId=7e06aaa7-6f16-4cfd-a231-8d0882829ef4

Doing that as a separate task would let us get the join links in quickly, which was the main point of the issue originally :D

File: packages/web/components/Player/Filter/PlayerFilter.tsx [hunk 2]
```diff
@@ -122,86 +124,286 @@ export const PlayerFilter: React.FC<Props> = ({
...
+                    base: 'sm',
+                    md: '1em',
+                  }}
+                  height="40px"
+                  minW={{
+                    base: '14em',
+                    md: '30em',
+                    '2xl': '20em',
+                  }}
+                  onChange={(e) => setSearch(e.target.value)}
```
[the remark this answers is not in the document]
Comment (HHH-GH) [reply packages/web/components/Player/Filter/PlayerFilter.tsx:295]: I would be willing to look at it, and I'd attempt to code it up unless it was way above my level 😅

File: packages/web/pages/guilds.tsx (hunk 1 above)
Comment (HHH-GH) [reply packages/web/pages/guilds.tsx:31]: But I can make it match the roles and raids pages more closely as part of this work, yes?

File: packages/web/components/Player/Filter/PlayerFilter.tsx (hunk 2 above)
[the remark this answers is not in the document]
Comment (vidvidvid) [reply packages/web/components/Player/Filter/PlayerFilter.tsx:295]: just add `onSearch(e.target.value)` into the function :D

File: packages/web/pages/guilds.tsx (hunk 1 above)
Comment (vidvidvid) [reply packages/web/pages/guilds.tsx:31]: yup, i'm talking about the part from your screenshot. roles and raids don't match, so aligning the spacings between all these pages would be really nice. but maybe we can wait for the designers to decide on the exact numbers

Comment (HHH-GH) [reply packages/web/pages/guilds.tsx]: I've made the spacing match - 32px at md and higher

**Review by vidvidvid — APPROVED** [review]
Looks great!

**Pull request merged**